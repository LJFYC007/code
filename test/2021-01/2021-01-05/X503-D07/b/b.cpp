/***************************************************************
	File name: b.cpp
	Author: ljfcnyali
	Create time: 2021年01月05日 星期二 09时03分26秒
***************************************************************/
#include<bits/stdc++.h>

using namespace std;

#define REP(i, a, b) for ( int i = (a), _end_ = (b); i <= _end_; ++ i ) 
#define mem(a) memset ( (a), 0, sizeof ( a ) ) 
#define str(a) strlen ( a ) 
#define pii pair<int, int>
typedef long long LL;

const int maxn = 1e5 + 10;

int n;
vector<pii> ans;

inline void Build(int now, int k)
{
    -- k; if ( k == 1 ) return ; 
    vector<int> p; p.clear();   
    for ( int i = 2; i * i <= k; ++ i ) while ( k % i == 0 ) { p.push_back(i); k /= i; } 
    if ( k > 1 ) p.push_back(k);

    for ( int i = 0; i < p.size(); ++ i ) 
    {
        ++ n; ans.push_back(pii(n, now));
        Build(n, p[i]);
    }
}

int main()
{
    freopen("b.in", "r", stdin);
    freopen("b.out", "w", stdout);
    int k;
    while ( ~scanf("%d", &k) ) 
    {
        ans.clear();
        n = 1; Build(1, k + 1);
        printf("%d\n", n);
        for ( int i = 0; i < ans.size(); ++ i ) printf("%d %d\n", ans[i].first, ans[i].second);
    }
    return 0;
}
