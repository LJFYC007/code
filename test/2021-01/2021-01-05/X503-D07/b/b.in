1
2
6
16
14007
998244353
