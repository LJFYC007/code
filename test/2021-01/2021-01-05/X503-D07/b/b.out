1
2
2 1
4
2 1
3 1
4 3
5
2 1
3 1
4 1
5 1
21
2 1
3 2
4 1
5 4
6 4
7 6
8 1
9 8
10 8
11 10
12 10
13 12
14 12
15 1
16 15
17 15
18 15
19 18
20 18
21 20
34
2 1
3 2
4 2
5 2
6 2
7 2
8 2
9 2
10 2
11 2
12 2
13 2
14 2
15 2
16 2
17 2
18 2
19 2
20 2
21 2
22 2
23 2
24 2
25 2
26 2
27 26
28 26
29 28
30 2
31 30
32 30
33 30
34 30
